### PR TITLE
add ZeroDivisionError exception to WaterOrientationRelaxation class

### PR DESCRIPTION
**Pull request by cyanezstange**
Hi!, I've been using waterdynamics analysis module and find and error in WaterOrientationRelaxation class, this is a by-zero-division that happens when the selection of atoms is too small.
I added a fix in the same way of the SurvivalProbability class at 5dcf427ad85cafcb84f2901ee629302473563bde.

### CODE REVIEW

File: package/MDAnalysis/analysis/waterdynamics.py [hunk 1]
```diff
@@ -639,7 +639,10 @@ def _getMeanOnePoint(self,universe,selection1,selection_str,dt,totalFrames):
         valdipList = []
 
         for j in range(totalFrames/dt-1):
-            a = self._getOneDeltaPoint(universe,repInd,j,sumsdt,dt)
+            try:
+                a = self._getOneDeltaPoint(universe,repInd,j,sumsdt,dt)
+            except ZeroDivisionError:
+                continue
```
Comment (orbeckst) [inline package/MDAnalysis/analysis/waterdynamics.py]: Is it really ok to just ignore the division by zero? Or should this be fixed elsewhere?

**Comment by orbeckst** [comment]
@cyanezstange also note that GitHub does not recognize the commit as coming from your GitHub account. Please add the email address that you are using to your GitHub profile.

File: package/MDAnalysis/analysis/waterdynamics.py (hunk 1 above)
Comment (orbeckst) [inline package/MDAnalysis/analysis/waterdynamics.py]: If you are sure that this is the best fix then please add at least a comment to the code explaining under which conditions you expect the exception to occur, e.g., elaborate on "happens when the selection of atoms is too small".

**Comment by orbeckst** [comment]
@cyanezstange please address the comments before we can merge. Thanks!

**Comment by orbeckst** [comment]
@alejob – do you have any comments on this patch? `analysis.waterdynamics` is your module. 

Without answers to the comments here we are not going to apply the patch and I haven't heard from @cyanezstange in almost a month.

**Comment by alejob** [comment]
Hi Oliver, I think that the solution is a good option, because the problem is generated when in time *t* and in time *t + dt* there aren't repeated molecules (and/or when there aren't molecules in the selection), so I can't evaluate time correlation. Usually in very small selections we have this problem that there aren't repeated molecules in *t* and *t + dt*. With ignoring the division by zero we are just removing one point from the mean. This final mean is one point in our output graphic. So, for instance, instead of calculate the mean with 1000 points, we would calculate this mean with 994 points (6 zero division). 

I think that a user shouldn't select a so small group of atoms to run this analysis, but if he decides to make it, we have this solution.

By the way, @cyanezstange is working in our lab, I'll comment this to him.

Best.

**Comment by cyanezstange** [comment]
Hi Oliver, sorry for the late answer,
@alejob and I have been  discussing the correctness of this fix the last weeks, and we think it's OK.
I added my other email to my GitHub profile and the comment explaining the code too.

That's all I think, thanks for all. Regards.

Carlos.

**Comment by orbeckst** [comment]
Many thanks!

**Pull request merged**